### PR TITLE
Music recommendation notebook fixes

### PR DESCRIPTION
**Pull request by ivanmkc**
1. Fixed broken links
2. Added instructions for JupyterLab environments
3. Added better parameter handling
4. Fixed broken library references
5. Fixed missing import for pyarrow by installing pandas extra for google-cloud-bigquery
6. Fixed all lint issues

### CODE REVIEW

**Comment by sirtorry** [comment]
now this is clean.

File: notebooks/samples/tables/music_recommendation/music_recommendation.ipynb [hunk 1]
```diff
@@ -360,7 +363,7 @@
         "  keyfile_upload = files.upload()\n",
         "  keyfile = list(keyfile_upload.keys())[0]\n",
         "  %env GOOGLE_APPLICATION_CREDENTIALS $keyfile\n",
-        "  ! gcloud auth activate-service-account --key-file $keyfile"
+        "  ! gcloud auth activate-service-account --key-file '$keyfile'"
```
Comment (ivanmkc) [inline notebooks/samples/tables/music_recommendation/music_recommendation.ipynb:366]: Needed to handle files with spaces in the name.

File: notebooks/samples/tables/music_recommendation/music_recommendation.ipynb [hunk 2]
```diff
@@ -243,7 +243,7 @@
       "source": [
         "# Use the latest major GA version of the framework.\n",
         "! pip install --upgrade --quiet --user google-cloud-automl \n",
-        "! pip install --upgrade --quiet --user google-cloud-bigquery\n",
+        "! pip install --upgrade --quiet --user 'google-cloud-bigquery[pandas]'\n",
```
Comment (ivanmkc) [inline notebooks/samples/tables/music_recommendation/music_recommendation.ipynb:246]: Needed quotes for it to work in non-colab environment

Comment (ivanmkc) [reply notebooks/samples/tables/music_recommendation/music_recommendation.ipynb:246]: We might think about adding single quotes to all install lines for consistency.

**Comment by ivanmkc** [comment]
I checked the CI and I think the error is expected. The error is related to the automated running of the notebook being blocked by:

`DefaultCredentialsError: File /path/to/service/account was not found.`

**Comment by ivanmkc** [comment]
Also there are a lot of existing notebook lint issues. @sirtorry should I manually fix them? Or is there some automated tool?

https://github.com/GoogleCloudPlatform/ai-platform-samples/pull/226/checks?check_run_id=1299290688

**Comment by sirtorry** [comment]
for service account CI issue, see: https://github.com/GoogleCloudPlatform/ai-platform-samples/blob/fcdae7b6622f3da94e197b8899c69b805956d10f/notebooks/templates/ai_platform_notebooks_template.ipynb

for lint, i've not automated fixing issues yet. you're welcome to figure out how to automate it or manually fix them.

**Comment by ivanmkc** [comment]
@sirtorry fixed all lint issues!

**Comment by ivanmkc** [comment]
@sirtorry can we just merge this for now since CI seems to have an issue?

It doesn't seem to respect the `no-remove` tag at the moment.

**Comment by ivanmkc** [comment]
If you're fine merging, I'll run nbfmt again (to remove the no-remove tags and the CI warning for nbfmt).

**Comment by sirtorry** [comment]
let's figure out a solution for auth and appease the notebook run CI or fix the notebook run CI. we can ignore notebook format CI for now.

**Pull request closed without merging**